azure-functions-durable/src/lib.rs

```diff
@@ -1,2 +1,401 @@
+    #[serde(rename = "sendEventPostUri")]
+    pub raise_event_url: String,
+    /// The "terminate" URL of the orchestration instance.
+    #[serde(rename = "terminatePostUri")]
+    pub terminate_url: String,
+    /// The "rewind" URL of the orchestration instance.
+    #[serde(rename = "rewindPostUri")]
+    pub rewind_url: String,
+    /// The "purge history" URL of the orchestration instance.
+    #[serde(rename = "purgeHistoryDeleteUri")]
+    pub purge_history_url: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DurableOrchestrationClientBinding {
+    #[serde(rename = "taskHubName")]
+    task_hub: String,
+    creation_urls: CreationUrls,
+    management_urls: ManagementUrls,
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub enum OrchestrationRuntimeStatus {
+    Running,
+    Pending,
+    Failed,
+    Canceled,
+    Terminated,
+    Completed
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub enum OrchestrationClientError
+{
+    //400
+    InstanceFailedOrTerminated,
+    //410
+    InstanceCompletedOrFailed,
+    //404
+    InstanceNotFound,
+    //400
+    BadRaiseEventContent,
+    //500
+    UnspecifiedError,
+    CommunicationError(String),
+    InvalidResponse,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct OrchestrationHistoryEvent {
+    event_type: String,
+    orchestration_status:Option<OrchestrationRuntimeStatus>,
+    function_name: Option<String>,
+    result: Option<Value>,
+    scheduled_time: Option<DateTime<Utc>>,
+    timestamp: DateTime<Utc>
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OrchestrationStatus {
+    runtime_status:OrchestrationRuntimeStatus,
+    input: Option<Value>,
+    custom_status: Option<Value>,
+    output: Option<Value>,
+    created_time: DateTime<Utc>,
+    history_events: Option<Vec<OrchestrationHistoryEvent>>
+}
+
+#[derive(Default, Builder, Debug)]
+#[builder(setter(strip_option),default)]
+pub struct InstanceQuery<'a> {
+    instance_id:Option<&'a str>,
+    created_time_from:Option<DateTime<Utc>>, 
+    created_time_to:Option<DateTime<Utc>>, 
+    runtime_status:Option<Vec<OrchestrationRuntimeStatus>>,
+    task_hub:Option<&'a str>,
+    connection_name:Option<&'a str>,
+    top:Option<u32>
+}
+
+impl<'a> InstanceQuery<'a> {
+    pub fn with_id(id:&str) -> InstanceQuery {
+        InstanceQueryBuilder::default().instance_id(id).build().unwrap()
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PurgeHistoryResult {
+    instances_deleted:u32
+}
+
+type OrchestrationResult<T> = Result<T, OrchestrationClientError>;
+
+pub struct OrchestrationClient {
+    binding: DurableOrchestrationClientBinding,
+    client:Client<HttpConnector>
+}
+
+impl OrchestrationClient {
+    pub fn new(binding:DurableOrchestrationClientBinding) -> OrchestrationClient {
```

Comment:
I think if we change this to take the `CreationUrls` and `ManagementUrls` (which the client stores internally instead of the binding), I'm 👍 on these changes.  Looking like a great start to the client!